src/stdlib/encode_charset.rs

```diff
@@ -0,0 +1,149 @@
+use crate::diagnostic::Label;
+use crate::prelude::*;
+use encoding_rs::Encoding;
+use nom::AsBytes;
+use std::str::from_utf8;
+
+#[derive(Clone, Copy, Debug)]
+pub struct EncodeCharset;
+
+impl Function for EncodeCharset {
+    fn identifier(&self) -> &'static str {
+        "encode_charset"
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "Encode charset to euc-kr",
+                source: r#"encode_base64(encode_charset!("안녕하세요", "euc-kr"))"#,
+                result: Ok("vsiz58fPvLy/5A=="),
+            },
+            Example {
+                title: "Encode charset to euc-jp",
+                source: r#"encode_base64(encode_charset!("こんにちは", "euc-jp"))"#,
+                result: Ok(r"pLOk86TLpMGkzw=="),
+            },
+            Example {
+                title: "Encode charset to gb2312",
+                source: r#"encode_base64(encode_charset!("你好", "gb2312"))"#,
+                result: Ok(r"xOO6ww=="),
+            },
+        ]
+    }
+
+    fn summary(&self) -> &'static str {
+        "Encode UTF-8 to non UTF-8 charset"
+    }
+
+    fn usage(&self) -> &'static str {
+        indoc! {"
+            Encode UTF-8 to non UTF-8 charset.
+
+            The `value` parameter is a UTF-8 encoded string.
+            The `to_charset` parameter specifies the charset to encode the `value`.
+        "}
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "to_charset",
+                kind: kind::BYTES,
+                required: true,
+            },
+        ]
+    }
+
+    fn compile(
+        &self,
+        _state: &TypeState,
+        _ctx: &mut FunctionCompileContext,
+        arguments: ArgumentList,
+    ) -> Compiled {
+        let value = arguments.required("value");
+        let to_charset = arguments.required("to_charset");
+
+        Ok(DecodeCharsetFn { value, to_charset }.as_expr())
+    }
+}
+
+fn encode_charset(value: &str, to_charset: &[u8]) -> Resolved {
+    let encoder = Encoding::for_label(to_charset).ok_or_else(|| error(to_charset))?;
+
+    let (output, _, _) = encoder.encode(value);
+    Ok(Value::Bytes(output.as_bytes().to_vec().into()))
+}
+
+fn error(from: &[u8]) -> ExpressionError {
```

Comment:
Nit: This is identical to the one above. Maybe we can move everything under a new directory `src/stdlib/charset`